cattle/plugins/docker/storage.py

```diff
@@ -19,41 +19,56 @@ def __init__(self):
     def _get_image_by_id(id):
         templates = docker_client().images(all=True)
         templates = filter(lambda x: x['Id'] == id, templates)
-
         if len(templates) > 0:
             return templates[0]
         return None
 
     @staticmethod
-    def _get_image_by_label(tag):
-        templates = docker_client().images(all=True, name=tag.split(':', 1)[0])
-        templates = filter(lambda x: tag in x['RepoTags'], templates)
-
+    def _get_image_by_label(uuid):
+        parsed_uuid = DockerPool.parse_repo_tag(uuid)
+        templates = docker_client().images(all=True, name=parsed_uuid['repo'])
+        templates = filter(lambda x: parsed_uuid['uuid'] in x['RepoTags'],
+                           templates)
         if len(templates) > 0:
             return templates[0]
         return None
 
     def pull_image(self, image, progress):
+        log.info('Image is: %s', image)
         if not self._is_image_active(image, None):
             self._do_image_activate(image, None, progress)
 
     def _is_image_active(self, image, storage_pool):
-        image_obj = self._get_image_by_label(image.data.dockerImage.fullName)
+        if isinstance(image, basestring):
+            image_obj = self._get_image_by_label(image)
+        else:
```

Comment:
Do you have a test case that exercises this else block? I think the answer to that is no since you pulled out the image.data object from the tests.

If that's true, you should add a test or two that covers this else block.